frontend/src/components/modals/CreateFolder.vue

```diff
@@ -0,0 +1,178 @@
+<template>
+  <div v-if="isModalVisible"
+       class="modal-backdrop"
+       tabindex="-1"
+       role="dialog"
+       aria-labelledby="createFolderModalLabel"
+       aria-hidden="true"
+       @keypress.enter="createFolder"
+       @click="close"
+  >
+    <div class="modal-dialog" role="document">
+      <div class="modal-content" @click.stop="">
+        <div class="modal-header">
+          <h5 class="modal-title font-weight-bold">Create a Folder</h5>
+          <button type="button"
+                  class="close"
+                  id="closeButton"
+                  data-toggle="tooltip"
+                  data-placement="right"
+                  title="Close"
+                  aria-label="Close"
+                  @click="close"
+          >
+            <span aria-hidden="true">&times;</span>
+          </button>
+        </div>
+        <div class="modal-body">
+          <div class="form-group">
+            <input v-model="folderName"
+                   type="text"
+                   class="form-control"
+                   ref="inputField"
+                   placeholder="Enter your folder name"
+                   @input="hideErrorMessage"
+            >
+            <p v-if='errorMessage !== ""' id="errorMessage">{{errorMessage}}</p>
+          </div>
+        </div>
+        <div class="modal-footer">
+          <button type="button" class="btn font-weight-bold" id="createButton" @click="createFolder">
+            Create
+          </button>
+        </div>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'CreateFolderModal',
+  props: ["isModalVisible"],
+  data() {
+    return {
+      folderName: "",
+      errorMessage: "",
+    }
+  },
+  updated() {
+    if (this.isModalVisible) {
+      window.addEventListener('keydown', this.escapeHandler);
```

Comment:
I think it would be cleaner to use `@keydown.escape="escapeHandler"` on the modal element itself and checking for the `this.isModalVisible` inside the event handler
especially since this will add the same eventListener again and again on every update and have it fire multiple times every time the modal gets opened.

Reply:
I tried this before adding the event listener and it didn't work as expected. The listener is removed when closing the modal, so there is always maximum one listener at the same time.